### PR TITLE
perf(orchestration): speed up new-chat propagation and offline catch-up

### PR DESCRIPTION
**Pull request by RusiruSadathana**
## What Changed

- Coalesces shell events per aggregate over a 50 ms / 512-event window with bounded shell-refetch concurrency. Bursty activity for one thread now collapses to one shell refetch without blocking unrelated new threads.
- Captures the authoritative shell head when a client resumes. Gaps larger than `SHELL_RESUME_MAX_GAP` (1,000), or cursors ahead of that head, receive a fresh projected snapshot.
- Replays small gaps only through the captured head. Events committed afterward are delivered by the already-attached live subscription, so catch-up cannot chase a moving event-store head while an unbounded live buffer grows.
- Starts the live PubSub consumer immediately before snapshot or replay work, preserving events published during a synchronous snapshot load.
- Keeps deletion/archive coalescing correct by emitting `thread-removed` when the current active-thread projection is absent.

## Why

New chats could be slow to appear because every high-frequency thread event triggered its own shell refetch, serializing unrelated thread creation behind redundant database reads. Offline catch-up could also replay an effectively unbounded event range before reaching the live stream.

This keeps replay and WebSocket delivery, but makes replay explicitly bounded and uses an authoritative snapshot when replay would be more expensive.

## Tests

Integration coverage verifies:

- large-gap and invalid-cursor snapshot fallback;
- replay is limited to the captured head;
- replay burst coalescing and deletion/removal behavior;
- actual post-synchronization `streamDomainEvents` burst coalescing;
- events published during a synchronous snapshot load are buffered before the synchronization marker;
- `OrchestrationEngine.latestSequence` follows the authoritative command read model.

## UI Changes

No rendered UI changes. Shell and thread status indicators added by #4163 now receive a bounded, explicit synchronization completion boundary.

## Checklist

- [x] This PR is focused
- [x] The synchronization and performance behavior is covered by integration tests
- [x] No screenshots or animation video are applicable

## Related Issues

Related to #2761 and #2750.

---

> [!NOTE]
> **Medium Risk**
> Changes real-time WebSocket shell synchronization and resume semantics (snapshot fallback, coalescing, removal emissions); behavior is heavily integration-tested but clients depend on ordering and sequence handling.
> 
> **Overview**
> Improves **`subscribeShell`** so sidebar/shell updates stay fast under bursts and after long offline periods.
> 
> The orchestration engine exposes **`latestSequence`** from the authoritative command read model. On resume, the handler compares the client cursor to that head: gaps over **1,000** events or cursors ahead of the head get a **fresh shell snapshot** instead of per-event replay; smaller gaps replay only through the captured head so catch-up cannot chase a moving store while the live buffer grows.
> 
> **Live subscription** now starts **before** snapshot or replay work, with events buffered and coalesced (50 ms / 512-event windows, **one refetch per aggregate** per batch, concurrency 8). High-frequency deltas on one thread no longer block unrelated **`thread.created`** updates. Shell projection refetches **retry once** on transient DB errors; a missing row after coalescing emits **`thread-removed`** / **`project-removed`** instead of silently dropping deletes.
> 
> Extensive **`server.test.ts`** coverage exercises snapshot fallback, bounded replay, coalescing, live bursts after sync, and deletion edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 465efdcd2dcbe7f630ce4003847f1f033ad0e085. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

> [!NOTE]
> ### Speed up new-chat propagation and offline catch-up in `subscribeShell`
> - Reworks the `subscribeShell` WebSocket RPC in [ws.ts](https://github.com/pingdotgg/t3code/pull/4177/files#diff-7f2100e8ffa23a58d9bf425c5b3ceafc39239911ae33bc569291a00ecb8e9125) to buffer live domain events before loading state, so no events are missed during startup.
> - Adds a `latestSequence` accessor to `OrchestrationEngineShape` so the handler can bound catch-up to the captured head sequence and avoid unbounded replay.
> - Introduces `SHELL_RESUME_MAX_GAP` (1,000 events): if the client's cursor is too far behind (or ahead), a fresh shell snapshot is sent instead of incremental replay.
> - Coalesces multiple events for the same aggregate into a single refetch during both catch-up (`coalesceShellEvents`) and live streaming (`coalesceShellLiveStream`), reducing redundant shell reads.
> - Thread and project events now emit a removal when the projected shell row is absent, rather than silently dropping the event.
> - `retryShellProjectionRead` retries a failing shell read once before returning `Option.none`, preventing transient errors from propagating as stream failures.
>
> 
>
> <sup><a href="https://app.macroscope.com">Macroscope</a> summarized 465efdc.</sup>
> 
>

### CODE REVIEW

File: apps/server/src/ws.ts [hunk 1]
```diff
@@ -1066,45 +1121,65 @@ const makeWsRpcLayer = (
...
+                        new OrchestrationGetSnapshotError({
+                          message: "Failed to load orchestration shell snapshot",
+                          cause,
+                        }),
+                    ),
+                  );
+                  return Stream.concat(
+                    Stream.make({ kind: "snapshot" as const, snapshot }),
+                    liveStream,
+                  );
```
[the remark this answers is not in the document]
Comment (RusiruSadathana) [reply apps/server/src/ws.ts:1299]: Fixed. The live stream is now attached into a scope-bound buffer before any snapshot load, and all three paths (large-gap fallback, incremental catch-up, and the no-afterSequence path) emit from that buffered live tail, so an event published while the snapshot query is in flight is no longer dropped. This mirrors the subscribe-before-snapshot approach used for `subscribeThread`.

File: apps/server/src/ws.ts [hunk 2]
```diff
@@ -678,6 +685,54 @@ const makeWsRpcLayer = (
...
+          for (const event of events) {
+            latestByAggregate.set(`${event.aggregateKind}:${event.aggregateId}`, event);
+          }
+          const survivors = Array.from(latestByAggregate.values()).sort(
+            (left, right) => left.sequence - right.sequence,
+          );
+          const shellEvents = yield* Effect.forEach(survivors, toShellStreamEvent, {
+            concurrency: SHELL_REFETCH_CONCURRENCY,
+          });
+          return shellEvents.flatMap((option) => (Option.isSome(option) ? [option.value] : []));
```
[the remark this answers is not in the document]
Comment (RusiruSadathana) [reply apps/server/src/ws.ts:781]: Fixed. `toShellStreamEvent` now emits `thread-removed` when `getThreadShellById` returns none instead of dropping the event, so when coalescing collapses a `thread.deleted`/`archived` into a later refetchable event for the same thread, the refetch returning none for the inactive row still surfaces the removal. A `thread-removed` for an entity the client does not hold is a harmless no-op. Added a test covering delete-then-later-event within one coalescing window.
